File: backend/src/main/java/com/yourcompany/onlineexam/model/Course.java
package com.yourcompany.onlineexam.model;

import lombok.Data;
import java.util.Date;
import java.util.List;

@Data
public class Course {
    private String id;  // Firestore dùng String ID
    private String code;
    private String name;
    private String description;
    private Integer credits;
    private String department;
    private Date createdAt;
    private Date updatedAt;
    private List<String> students; // Danh sách userId hoặc email sinh viên

    public void setId(String id) {
        this.id = id;
    }

    public List<String> getStudents() {
        return students;
    }

    public void setStudents(List<String> students) {
        this.students = students;
    }
}

File: backend/OnlineExam/Models/Course.cs
using Google.Cloud.Firestore;

namespace OnlineExam.Models;

[FirestoreData]
public class Course
{
    [FirestoreDocumentId]
    public string? Id { get; set; }
    [FirestoreProperty("code")]
    public string? Code { get; set; }
    [FirestoreProperty("name")]
    public string? Name { get; set; }
    [FirestoreProperty("description")]
    public string? Description { get; set; }
    [FirestoreProperty("credits")]
    public int? Credits { get; set; }
    [FirestoreProperty("department")]
    public string? Department { get; set; }
    [FirestoreProperty("createdAt")]
    public Timestamp? CreatedAt { get; set; }
    [FirestoreProperty("updatedAt")]
    public Timestamp? UpdatedAt { get; set; }
    [FirestoreProperty("students")]
    public List<string>? Students { get; set; }
}
